src/coreclr/src/vm/codeman.cpp

```diff
@@ -6778,6 +6778,12 @@ BOOL ReadyToRunJitManager::JitCodeToMethodInfo(RangeSection * pRangeSection,
 
     // READYTORUN: FUTURE: Hot-cold spliting
 
+    // If the address is in a thunk, return NULL.
+    if (GetStubCodeBlockKind(pRangeSection, currentPC) != STUB_CODE_BLOCK_UNKNOWN)
```

Comment:
It may be a good idea to cache the result of `pReadyToRunInfo->FindSection` in ReadyToRunInfo now that `GetStubCodeBlockKind` will be called much more often.

Reply:
I've done that, and marked the ReadyToRunJitManager as a final class. The end result is that there is a minimal amount of extra code here now.